client/graphite/read.go

```diff
@@ -109,15 +120,23 @@ func (c *Client) queryToTargetsWithTags(ctx context.Context, query *prompb.Query
 	return targets, nil
 }
 
-func (c *Client) filterTargets(query *prompb.Query, targets []string) ([]string, error) {
+func (c *Client) filterTargets(logger log.Logger, query *prompb.Query, targets []string, r *http.Request) ([]string, error) {
 	// Filter out targets that do not match the query's label matcher
 	var results []string
+
+	graphitePrefix, err := c.fetchGraphitePrefix(logger, r)
+	if err != nil {
+		level.Debug(c.logger).Log(
```

Comment:
Why are you passing the logger ? (you can use c.logger)

Reply:
I removed it. I need to push a new patchset but I would like to finish the tests first.